### DISCUSSION TITLE
how to capture data from use-places-autocomplete?

### DISCUSSION THREAD

**Discussion by ReformedCola** [react-hook-form/react-hook-form#6149] [Q&A]
CSB: https://codesandbox.io/s/chakra-ui-header-forked-3gz0n?file=/src/example.js

Hi everyone, I wonder how to capture the data from the input box with use-places-autocomplete?

**Comment by bluebill1049** [comment]
Have u tried with Controller?

**Reply by ReformedCola** [reply]
@bluebill1049 Yes, the example I gave above is using Controller.

**Reply by ReformedCola** [reply]
Oops, sorry! I used it in the opposite way.
Here is the updated CSB: https://codesandbox.io/s/chakra-ui-header-forked-3gz0n?file=/src/PlacesAutocomplete.js
After using the `value` and `onChange` from `field`, it couldn't trigger the use-places-autocomplete, what should I do to trigger the autocomplete and capture the value to the form at the same time?

**Reply by nonoumasy** [reply]
@ReformedCola @bluebill1049 
I'm also trying to use RHF with UPA..

Still new to RHF 7. I got simple inputs like TextField to work but when working with other libraries such as use-places-autocomplete, I'm still unsure how to use the Controller effectively. I think the problem is that somehow the value returned by use-places-autocomplete is not wired up to RHF so its not returning it as part of the RHF data object.
I think this is because onChange is using the handleInput from UPA. How do I wire this up so that whatever handleInput returns is passed to RHF onChange?

Someone suggested:
"Your TextField is getting its value from useAutoComplete and not from RHF itself." so how would I pass it to RHF. Would I need to setup a state for this and return that seperately?

https://codesandbox.io/s/rhf-7-with-reusable-component-m9njl?file=/src/components/GooglePlacesAutocomplete.js:64-87

```
<Controller
  name="storyLocation"
  control={control}
  rules={{ required: 'storyLocation is Required' }}
  defaultValue=""
  render={({ field: { onChange, ref, name } }) => (
    <TextField
      fullWidth
      type="text"
      label={name}
      margin="normal"
      variant="outlined"
      ref={ref}
      value={value}
      onChange={handleInput}
      disabled={!ready}
      error={Boolean(errors && errors[name])}
      helperText={errors && errors[name]?.message}
    />
  )}
/>
```

should I do something like
```
render={({ field: { onChange, ref, name } }) => (
    <TextField
    {...rest}
      onChange={(e) => onChange(setValue(e.target.value)}
    />
  )}
  ```

**Reply by ReformedCola** [reply]
@nonoumasy @bluebill1049
I haven't figured it out yet. I've tried the way you prvovided but it was not working.
Do you have any working solutions yet?
